reflect/src/main/java/dagger/reflect/TypeUtil.java

```diff
@@ -86,7 +86,9 @@ static void checkNotPrimitive(Type type) {
                 "unexpected owner type for " + rawType + ": " + ownerType);
           }
         } else if (enclosingClass != null) {
-          throw new IllegalArgumentException("unexpected owner type for " + rawType + ": null");
+          // TODO ?
```

Comment:
not sure if this is the right thing to do here, but the nested generic test fails when this exception is thrown.

Reply:
What are the input values?

Reply:
```
java.lang.IllegalArgumentException: unexpected owner type for class com.example.JustInTimeGenericNested$JobProvider: null
	at dagger.reflect.TypeUtil$ParameterizedTypeImpl.<init>(TypeUtil.java:90)
	at dagger.reflect.UnlinkedJustInTimeBinding.findKeyForParameterizedType(UnlinkedJustInTimeBinding.java:78)
	at dagger.reflect.UnlinkedJustInTimeBinding.getTypeKeyForParameter(UnlinkedJustInTimeBinding.java:50)
	at dagger.reflect.UnlinkedJustInTimeBinding.link(UnlinkedJustInTimeBinding.java:37)
	at dagger.reflect.Linker.link(Linker.java:35)
	at dagger.reflect.Scope.link(Scope.java:143)
	at dagger.reflect.Scope.putJitBinding(Scope.java:136)
	at dagger.reflect.Scope.findBinding(Scope.java:76)
	at dagger.reflect.Scope.getBinding(Scope.java:39)
	at dagger.reflect.ComponentInvocationHandler.createMethodInvocationHandler(ComponentInvocationHandler.java:91)
	at dagger.reflect.ComponentInvocationHandler.invoke(ComponentInvocationHandler.java:56)
	at com.example.$Proxy173.thing(Unknown Source)
	at com.example.IntegrationTest.justInTimeGenericNested(IntegrationTest.java:187)
```